cclib/parser/molcasparser.py

```diff
@@ -656,7 +663,18 @@ def extract(self, inputfile, line):
                             if float(i) != 0:
                                 homos += 1
 
-                    line = next(inputfile)
+                    aonames = []
+                    if line.split():
+                        if line.split()[0] == '1':
+                            while line.split() != [] and line.split()[0] != '--':
+                                aonames.append("{atom}_{orbital}".format(atom=line.split()[1], orbital=line.split()[2]))
+                                info = line.split()[3:]
```

Comment:
There is a lot of splitting going on here. Using it like this hurts readability and for big sections can hurt readability. Try and store/reuse the contents of a split more. Maybe something like:
```python
tokens = line.split()
if tokens and tokens[0] == '1':
    while tokens and tokens[0] != '--':
...
        line = next(inputfile)
        tokens = line.split()
```